spec/Section 5 -- Validation.md

```diff
@@ -564,6 +564,47 @@ fragment conflictingDifferingResponses on Pet {
 }
 ```
 
+The same is true if a field is designated `Non-Nullable` in an operation. In
+this case, `nickname` could be either a `String` or a `String!` which are two
+different types and therefore can not be merged:
+
+```graphql counter-example
+fragment conflictingDifferingResponses on Pet {
+  ... on Dog {
+    nickname
+  }
+  ... on Cat {
+    nickname!
+  }
+}
+```
+
+### Client Controlled Nullability Designator List Dimensions
+
+**Formal Specification**
+
+- For each {field} in the document
+  - Let {fieldDef} be the definition of {field}
+  - Let {fieldType} be the type of {fieldDef}
+  - Let {requiredStatus} be the required status of {field}
+  - Let {designatorDepth} be the number of ListNullability operators in
+    {requiredStatus}
+  - If {designatorDepth} is 0
+    - return true
+  - Let {typeDepth} be the number of list dimensions in {fieldType}
+  - If {typeDepth} equals {designatorDepth} or {designatorDepth} equals 0 return
+    true
```

Comment:
What about this:
```suggestion
  - If {typeDepth} is greater or equal to {designatorDepth} return true
```
?

This way, `0` is just a particular case of a more generic rule that allows "skipping over the list items you want unchanged"

With this schema:
```graphql
type Query {
  field: [[String]]
}
```

This would all be valid:
```graphql
{
  # make the list required
  a: field! 
  # make the items of the list required
  a: field[!] 
  # make the items of the items of the list required
  a: field[[!]] 
}
```

I don't think it's going to be used that much but it's a more generic rule and avoids "0" as an outlier: everything that's not mentioned explicitely is untouched

Reply:
Someone will need to got back through the notes, but IIRC the concern with this was that it's not clear how to read a list operator with fewer dimensions than the list type it's being applied to. ie 

Does `[!] == [[!]]` or does `[!] == [[]!]`? @calvincestari will be leading a CCN discussion at the next working group meeting on August 3rd, and we'll also be having a CCN working group meeting on the 26th if you'd like to join that to discuss more: https://github.com/graphql/client-controlled-nullability-wg/blob/main/agendas/2023/2023-07-26.md

Reply:
> Does `[!] == [[!]]` or does `[!] == [[]!]` ?

I think it would work like for a single dimension list. 

For this field definition `list1: [String]`, `list1!` is the equivalent of `list1[]!`, not `list1[!]` so it's just a generalisation of the "single list" case. Just like `! == []!`, I would expect `[!] == [[]!]` (modulo the [caveat from above](https://github.com/graphql/graphql-spec/pull/895#pullrequestreview-1540884946) about [] but that's something else).

> @calvincestari will be leading a CCN discussion at the next working group meeting on August 3rd, and we'll also be having a CCN working group meeting on the 26th if you'd like to join that to discuss more

I won't be able to join unfortunately but I'm working with Calvin so I'll discuss this with him :)!

Reply:
I also think `[!] == [[]!]` makes sense.

But `If {typeDepth} equals {designatorDepth} or {designatorDepth} equals 0 return` allows future relaxing the rule to be  `If {typeDepth} is greater or equal to {designatorDepth} return true` without breaking queries. So would it be possible to advance as is and wait for the feedback from the community?